### DISCUSSION TITLE
Modal Method 1 - ID.showModal() --- okay but what's the JS?

### DISCUSSION THREAD

**Discussion by noconsulate** [saadeghi/daisyui#1989] [Q&A] [answered]
Sorry if this is a dumb question but I can't figure out and the doc doesn't offer much of a clue. 

> opens on click using ID.showModal() method. can be closed using ID.close() method

What should the ID.close() method do actually? I don't mind using methods 2 or 3 but since this one is recommended I'd prefer to make it work.

**Comment by dominikkrulak** [comment]
The JS is up to your logic which you can write it yourself or use some open source script to do it for you.

**Comment by saadeghi** [comment, marked as answer]
You can use `<button onclick="MY_MODAL.showModal()">` to open `<dialog id="MY_MODAL">`  

And to close it, there are 2 ways:
- Having a button inside `<form method="dialog">` inside the `<dialog>`
- Using a button like `<button onclick="MY_MODAL.close()">` anywhere else. Clicking it will close the modal

It's just the native HTML element `<dialog>` and the way it opens/closes has nothing to do with daisyUI.

Read more: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog

**Reply by noconsulate** [reply]
Got i! Thank you. I I should have looked into the <dialog> element. 

edit: I haven't worked it out on my page yet, but I don't understand how the JavaScript works yet. Your clue to look into <dialog> will let me figure it out.
